### PR TITLE
feat(config): add zw141

### PR DESCRIPTION
**Pull request by abmantis**
Imported from OZW without changes.

### CODE REVIEW

**Comment by blhoward2** [comment]
This is in our todo list. #1600. If you want to do the legwork to get it in shape I'll deal with the conflict, otherwise it'll have to wait until I'm to those devices, which will hopefully be soon. We don't display all of the parameters if they aren't supported so you'll need to run down if there are any changes. And edit the labels to make them more clean. Get rid of unnecessary descriptions. Add units as appropriate. Split bitmasks if any. Etc.

File: packages/config/packages/config/config/devices/0x0371/zw141.json [hunk 1]
```diff
@@ -0,0 +1,503 @@
...
+		},
+		"2": {
+			"label": "Retransmit",
+			"maxNodes": 5
+		}
+	},
+	"paramInformation": {
+		"20": {
+			"label": "Motor behavior after powering on",
+			"description": "Set the motor behavior after powering on",
```
Comment (blhoward2) [inline packages/config/packages/config/config/devices/0x0371/zw141.json]: Remove unnecessary or duplicative descriptions (including the key).

File: packages/config/packages/config/config/devices/0x0371/zw141.json [hunk 2]
```diff
@@ -0,0 +1,503 @@
...
+			"minValue": 0,
+			"maxValue": 2,
+			"defaultValue": 0,
+			"unsigned": true,
+			"readOnly": false,
+			"writeOnly": false,
+			"allowManualEntry": false,
+			"options": [
+				{
+					"label": "Stay the same position",
```
Comment (blhoward2) [inline packages/config/packages/config/config/devices/0x0371/zw141.json]: Change to "Same position"

File: packages/config/packages/config/config/devices/0x0371/zw141.json [hunk 3]
```diff
@@ -0,0 +1,503 @@
...
+			"label": "Motor direction",
+			"description": "Set to toggle the motor running direction",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 255,
+			"defaultValue": 2,
+			"unsigned": true,
+			"readOnly": false,
+			"writeOnly": false,
+			"allowManualEntry": false
```
Comment (blhoward2) [inline packages/config/packages/config/config/devices/0x0371/zw141.json]: This description suggests there are only two options. You're going to have to track the manual down and add the options. This is already set to allowManualEntry: false, so it is effectively unusable at the moment. Make sure you fix the maxValue once you identify the real values.

File: packages/config/packages/config/config/devices/0x0371/zw141.json [hunk 4]
```diff
@@ -0,0 +1,503 @@
...
+		},
+		"34": {
+			"label": "Blade turn time",
+			"description": "Set the blade turn time for Venetian mode. Details can be found in section 4.2 of Advanced information Product Manual.",
```
Comment (blhoward2) [inline packages/config/packages/config/config/devices/0x0371/zw141.json]: Don't reference the manual. Either put in a description or omit it if unneeded.

File: packages/config/packages/config/config/devices/0x0371/zw141.json [hunk 5]
```diff
@@ -0,0 +1,503 @@
...
+			"valueSize": 2,
+			"unit": "seconds",
+			"minValue": 10,
+			"maxValue": 32767,
```
Comment (blhoward2) [inline packages/config/packages/config/config/devices/0x0371/zw141.json:71]: This maxValue suggests the unit isn't actually seconds. It must be ms or something. Please verify and change.

File: packages/config/packages/config/config/devices/0x0371/zw141.json [hunk 6]
```diff
@@ -0,0 +1,503 @@
...
+			"writeOnly": false,
+			"allowManualEntry": true
+		},
+		"35": {
+			"label": "Curtain trip time",
+			"description": "Set the run time of the curtain. Details can be found in section 4.2 of Advanced information Product Manual.",
+			"valueSize": 2,
+			"unit": "seconds",
+			"minValue": 500,
+			"maxValue": 32767,
```
Comment (blhoward2) [inline packages/config/packages/config/config/devices/0x0371/zw141.json:84]: Same as above. This isn't in seconds.

File: packages/config/packages/config/config/devices/0x0371/zw141.json [hunk 7]
```diff
@@ -0,0 +1,503 @@
...
+			"minValue": 0,
+			"maxValue": 2,
+			"defaultValue": 0,
+			"unsigned": true,
+			"readOnly": false,
+			"writeOnly": true,
+			"allowManualEntry": false,
+			"options": [
+				{
+					"label": "Set: Exit calibration mode/Report: Not in calibration mode",
```
Comment (blhoward2) [inline packages/config/packages/config/config/devices/0x0371/zw141.json]: These labels can only be a few words long at most. Please shorten them and make them more clear.

File: packages/config/packages/config/config/devices/0x0371/zw141.json [hunk 8]
```diff
@@ -0,0 +1,503 @@
...
+					"value": 1
+				},
+				{
+					"label": "Set: Enter Venetian mode calibration/Report: in Venetian mode calibration",
+					"value": 2
+				}
+			]
+		},
+		"37": {
+			"label": "User confirmation for calibration",
```
Comment (blhoward2) [inline packages/config/packages/config/config/devices/0x0371/zw141.json]: Please make this label more clear, then lost the description.

File: packages/config/packages/config/config/devices/0x0371/zw141.json [hunk 9]
```diff
@@ -0,0 +1,503 @@
...
+					"value": 0
+				},
+				{
+					"label": "Confirm calibration",
+					"value": 1
+				}
+			]
+		},
+		"38": {
+			"label": "Calibration mode status report",
```
Comment (blhoward2) [inline packages/config/packages/config/config/devices/0x0371/zw141.json]: Same. Also labels should be in Title Case. Descriptions and option values in Sentence case. Please carry throughout.

File: packages/config/packages/config/config/devices/0x0371/zw141.json [hunk 10]
```diff
@@ -0,0 +1,503 @@
...
+			"readOnly": true,
+			"writeOnly": false,
+			"allowManualEntry": false,
+			"options": [
+				{
+					"label": "Calibration mode complete",
+					"value": 0
+				},
+				{
+					"label": "Start calibration mode, going to reference point A",
```
Comment (blhoward2) [inline packages/config/packages/config/config/devices/0x0371/zw141.json]: Please shorten to the extent that you can.

File: packages/config/packages/config/config/devices/0x0371/zw141.json [hunk 11]
```diff
@@ -0,0 +1,503 @@
...
+					"value": 0
+				},
+				{
+					"label": "Venetian mode",
+					"value": 1
+				}
+			]
+		},
+		"40": {
+			"label": "Repositioning",
```
Comment (blhoward2) [inline packages/config/packages/config/config/devices/0x0371/zw141.json]: Unclear. Please clarify what this does and make the label more clear. If necessary add a description.

File: packages/config/packages/config/config/devices/0x0371/zw141.json [hunk 12]
```diff
@@ -0,0 +1,503 @@
...
+					"value": 0
+				},
+				{
+					"label": "In repositioning",
+					"value": 1
+				}
+			]
+		},
+		"42": {
+			"label": "Calibration lock",
```
Comment (blhoward2) [inline packages/config/packages/config/config/devices/0x0371/zw141.json]: Make the label "Enable / Disable Calibration Lock". Then lose the description.

File: packages/config/packages/config/config/devices/0x0371/zw141.json [hunk 13]
```diff
@@ -0,0 +1,503 @@
...
+					"value": 0
+				},
+				{
+					"label": "Enable calibration",
+					"value": 1
+				}
+			]
+		},
+		"48": {
+			"label": "S1 scene activation",
```
Comment (blhoward2) [inline packages/config/packages/config/config/devices/0x0371/zw141.json]: Change this to S1 Scene Activation Mode. Then change the option values to just "Key pressed one time," etc. Do the same for S2.

File: packages/config/packages/config/config/devices/0x0371/zw141.json [hunk 14]
```diff
@@ -0,0 +1,503 @@
...
+				},
+				{
+					"label": "Switch multilevel report",
+					"value": 2
+				}
+			]
+		},
+		"85": {
+			"label": "External button mode operation",
+			"description": "Set operation mode of external button. \n        In Operation Mode 1, external button 1 maps to z-wave button, external button 2 opens or stops motor.\n        In Operation Mode 2, external button 1 maps to z-wave button, external button 2 closes or stops motor.",
```
Comment (blhoward2) [inline packages/config/packages/config/config/devices/0x0371/zw141.json]: Remove tabs and newlines from descriptions.

File: packages/config/packages/config/config/devices/0x0371/zw141.json [hunk 15]
```diff
@@ -0,0 +1,503 @@
...
+			"minValue": 0,
+			"maxValue": 1,
+			"defaultValue": 0,
+			"unsigned": true,
+			"readOnly": false,
+			"writeOnly": false,
+			"allowManualEntry": false,
+			"options": [
+				{
+					"label": "Operation Mode 1",
```
Comment (blhoward2) [inline packages/config/packages/config/config/devices/0x0371/zw141.json]: Please figure out what these are and describe them.

File: packages/config/packages/config/config/devices/0x0371/zw141.json [hunk 16]
```diff
@@ -0,0 +1,503 @@
...
+					"value": 3
+				},
+				{
+					"label": "Automatic identification mode",
+					"value": 4
+				}
+			]
+		},
+		"243": {
+			"label": "DSK",
```
Comment (blhoward2) [inline packages/config/packages/config/config/devices/0x0371/zw141.json]: What is a DSK?

File: packages/config/packages/config/config/devices/0x0371/zw141.json [hunk 17]
```diff
@@ -0,0 +1,503 @@
...
+			"minValue": 1,
+			"maxValue": 3,
+			"defaultValue": 1,
+			"unsigned": true,
+			"readOnly": false,
+			"writeOnly": false,
+			"allowManualEntry": false,
+			"options": [
+				{
+					"label": "Enable network inclusion/exclusion",
```
Comment (blhoward2) [inline packages/config/packages/config/config/devices/0x0371/zw141.json:456]: These options don't seem to match the label. Please verify and fix.

File: packages/config/packages/config/config/devices/0x0371/zw141.json [hunk 18]
```diff
@@ -0,0 +1,503 @@
...
+			"readOnly": false,
+			"writeOnly": true,
+			"allowManualEntry": false,
+			"options": [
+				{
+					"label": "Reset to factory default setting",
+					"value": 0
+				},
+				{
+					"label": "Reset to factory default setting and remove from the z-wave network",
```
Comment (blhoward2) [inline packages/config/packages/config/config/devices/0x0371/zw141.json]: Please shorten. Maybe "Reset to factory and remove from Z-Wave"

**Comment by blhoward2** [comment]
I'm confused. We already support this device. A file already exists.

**Review by blhoward2 — CHANGES_REQUESTED** [review]
This file is in our todo list #1600, but if you want to give it a go I'll incorporate your label changes and such into the file. You'll need it to be completely finished and merged by tomorrow, though, as I plan to work on these this weekend.

I had a few minute so I took an initial run through. It may help to look at one of my more recent PRs to see the style we are going for. Our goal is to make these all as consistent as we can. That's why we're templating them as we go. Don't worry about templating them for now if you attempt this.

File: packages/config/packages/config/config/devices/0x0371/zw141.json (hunk 3 above)
Comment (abmantis) [reply packages/config/packages/config/config/devices/0x0371/zw141.json]: Fixed according to  https://help.aeotec.com/support/solutions/articles/6000198854-nano-shutter-user-guide-

File: packages/config/packages/config/config/devices/0x0371/zw141.json (hunk 8 above)
Comment (abmantis) [reply packages/config/packages/config/config/devices/0x0371/zw141.json]: I have no clue about this one. The manual doesn't say much.

**Comment by abmantis** [comment]
I'm glad to see that you are enforcing good config files for zwavejs! There were indeed some weird stuff on OZW files.
I've changed them as suggested. Let me know if I should do anything further.

**Comment by blhoward2** [comment]
Thanks! There were a few more minor changes but I just made them myself. Appreciate your help.

**Review by blhoward2 — DISMISSED** [review]

**Comment by blhoward2** [comment]
I've merged this into my upstream work, so I'm closing it here. This will merge with my template push.

**Comment by abmantis** [comment]
@blhoward2 what is the branch where you've merged this into? Github is not telling me, and I would like to track it to know when this gets release :)

**Comment by blhoward2** [comment]
It’s merged into my local branch. It’ll be a few days as I need to finish cleaning up and templating all of the Aeotec devices. You’ll see me push up a PR templating all of Aeotec.

**Comment by abmantis** [comment]
@blhoward2 any plans on getting this into the repo?

**Comment by abmantis** [comment]
@blhoward2 Its been two months already. Can you just merge this into master without the rest of the changes you have on your local branch?

**Comment by blhoward2** [comment]
No but the good news is that I'm aiming to merge this one today or tomorrow anyway.

**Pull request closed without merging**